components/layout/fragment.rs

```diff
@@ -2231,8 +2231,8 @@ impl Fragment {
                     offset -= minimum_line_metrics.space_needed().scale_by(percentage)
                 }
                 vertical_align::T::LengthOrPercentage(LengthOrPercentage::Calc(formula)) => {
-                    offset -= minimum_line_metrics.space_needed().scale_by(formula.percentage()) +
-                        formula.length()
+                    offset -= formula.to_computed(Some(minimum_line_metrics.space_needed()))
```

Comment:
Heh, these cases are the ones why I wanted the extra method :).

Probably not worth the effort though.